libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -526,7 +527,8 @@ void _IotMqtt_InvokeSubscriptionCallback( _mqttConnection_t * pMqttConnection,
         if( pSubscription->unsubscribed == true )
         {
             /* An unsubscribed subscription should have been removed from the list. */
-            IotMqtt_Assert( IotLink_IsLinked( &( pSubscription->link ) ) == false );
+            linkStatus = IotLink_IsLinked( &( pSubscription->link ) );
```

Comment:
Similarly, linkStatus will not be used when asserts are disabled.  Strange that the build check does not fail this...